api/src/main/java/org/apache/cloudstack/api/command/user/iso/DetachIsoCmd.java

```diff
@@ -48,6 +48,9 @@
             required=true, description="The ID of the virtual machine")
     protected Long virtualMachineId;
 
+    @Parameter(name=ApiConstants.FORCED, type=CommandType.BOOLEAN, description="If true, ejects the iso before detaching on vmware. Default: false", since = "4.15.1")
```

Comment:
@Pearl1594 What does force detach do exactly? I agree force attach - which ejects ISO before attaching any ISO.

Reply:
@sureshanaparti  It is noticed that when we attempt a detach operation on an iso attached to a VM, despite the cdrom device not being in use (not mounted) - it is locked, as a result of which a detach operation leads to the same error as mentioned in the PR description. Thus there is a need to add a force flag to proceed

Reply:
@Pearl1594 how about try ejecting without force, for detach ?